audiomentations/core/composition.py

```diff
@@ -19,59 +20,134 @@ def randomize_parameters(self, *args, **kwargs):
-        super(Compose, self).__init__(transforms, p, shuffle)
+        super().__init__(transforms, p, shuffle)
 
     def __call__(self, samples, sample_rate):
         transforms = self.transforms.copy()
-        if random.random() < self.p:
+        should_apply = random.random() < self.p
+        # TODO: Adhere to self.are_parameters_frozen
+        # https://github.com/iver56/audiomentations/issues/135
+        if should_apply:
             if self.shuffle:
                 random.shuffle(transforms)
             for transform in transforms:
                 samples = transform(samples, sample_rate)
 
         return samples
 
-    def randomize_parameters(self, samples, sample_rate):
-        for transform in self.transforms:
-            transform.randomize_parameters(samples, sample_rate)
-
 
 class SpecCompose(BaseCompose):
     def __init__(self, transforms, p=1.0, shuffle=False):
         super().__init__(transforms, p, shuffle)
 
     def __call__(self, magnitude_spectrogram):
         transforms = self.transforms.copy()
-        if random.random() < self.p:
+        should_apply = random.random() < self.p
+        # TODO: Adhere to self.are_parameters_frozen
+        # https://github.com/iver56/audiomentations/issues/135
+        if should_apply:
             if self.shuffle:
                 random.shuffle(transforms)
             for transform in transforms:
                 magnitude_spectrogram = transform(magnitude_spectrogram)
 
         return magnitude_spectrogram
 
-    def randomize_parameters(self, magnitude_spectrogram):
-        for transform in self.transforms:
-            transform.randomize_parameters(magnitude_spectrogram)
\ No newline at end of file
+
+class OneOf(BaseCompose):
+    """
+    OneOf randomly picks one of the given transforms when called, and applies that
+    transform.
+
+    Example usage:
+
+    ```
+    augment = OneOf([
+        TimeStretch(min_rate=0.8, max_rate=1.25, p=1.0),
+        PitchShift(min_semitones=-4, max_semitones=4, p=1.0),
+    ])
+
+    # Generate 2 seconds of dummy audio for the sake of example
+    samples = np.random.uniform(low=-0.2, high=0.2, size=(32000,)).astype(np.float32)
+
+    # Augment/transform/perturb the audio data
+    augmented_samples = augment(samples=samples, sample_rate=16000)
+
+    # Result: The audio was either time-stretched or pitch-shifted, but not both
+    ```
+    """
+
+    def __init__(self, transforms, p: float = 1.0):
+        super().__init__(transforms, p)
+        self.transform_index = 0
+        self.should_apply = True
+
+    def randomize_parameters(self, *args, **kwargs):
+        super().randomize_parameters(*args, **kwargs)
+        self.should_apply = random.random() < self.p
+        if self.should_apply:
+            self.transform_index = random.randint(0, len(self.transforms) - 1)
+
+    def __call__(self, *args, **kwargs):
+        if not self.are_parameters_frozen:
+            self.randomize_parameters(apply_to_children=False)
```

Comment:
So when `apply_to_children` is set to False, what does it mean here ? That you randomize parameterst to the only transform which will be applied ?

Reply:
It means that it only randomize the parameters (self.should_apply and self.transform_index) of "self", and not the parameters in the list self.transforms. The parameters in each transform will be randomized when it gets called.